kong2kic/testdata/custom_resources/json/output-expected.json

```diff
@@ -0,0 +1,835 @@
+                "healthy": {
+                    "http_statuses": [
+                        200,
+                        201,
+                        202,
+                        203,
+                        204,
+                        205,
+                        206,
+                        207,
+                        208,
+                        226,
+                        300,
+                        301,
+                        302,
+                        303,
+                        304,
+                        305,
+                        306,
+                        307,
+                        308
+                    ],
+                    "successes": 0
+                },
+                "type": "http",
+                "unhealthy": {
+                    "http_failures": 0,
+                    "http_statuses": [
+                        429,
+                        500,
+                        503
+                    ],
+                    "tcp_failures": 0,
+                    "timeouts": 0
+                }
+            },
+            "threshold": 0
+        },
+        "hash_on": "none",
+        "hash_fallback": "none",
+        "hash_on_cookie_path": "/"
+    },
+    "proxy": {
+        "protocol": "http",
+        "path": "/v1",
+        "retries": 5,
+        "connect_timeout": 5000,
+        "read_timeout": 60000,
+        "write_timeout": 60000
+    }
+}{
+    "kind": "KongIngress",
+    "apiVersion": "configuration.konghq.com/v1",
+    "metadata": {
+        "name": "example-service--route-thisisaveryvery-long2093a020ca",
+        "creationTimestamp": null,
+        "annotations": {
+            "kubernetes.io/ingress.class": "kong"
+        }
+    },
+    "route": {
+        "methods": [
+            "GET",
+            "POST"
+        ],
+        "headers": {
+            "x-another-header": [
+                "first-header-value",
+                "second-header-value"
+            ],
+            "x-my-header": [
+                "~*foos?bar$"
+            ]
+        },
+        "protocols": [
+            "http",
+            "https"
+        ],
+        "regex_priority": 1,
+        "strip_path": false,
+        "preserve_host": true,
+        "https_redirect_status_code": 302,
+        "snis": [
+            "example.com"
+        ]
+    }
+}{
+    "kind": "KongPlugin",
+    "apiVersion": "configuration.konghq.com/v1",
+    "metadata": {
+        "name": "example-service-rate-limiting-advanced",
+        "creationTimestamp": null,
+        "annotations": {
+            "kubernetes.io/ingress.class": "kong"
+        }
+    },
+    "config": {
+        "hide_client_headers": false,
+        "identifier": "consumer",
+        "limit": [
+            5
+        ],
+        "namespace": "example_namespace",
+        "strategy": "local",
+        "sync_rate": -1,
+        "window_size": [
+            30
+        ]
+    },
+    "plugin": "rate-limiting-advanced",
+    "status": {}
```

Comment:
I think we should make sure that no `status` or `metadata.creationTimestamp` fields (or any other of that kind) are part of the output. These are fields that are meant to be set by the Kubernetes API server or the owning controller. 

I imagine this could be done generically for every object serialized like so:
```go
func SerializeObjectDropping(obj interface{}, fieldsToDrop ...string) ([]byte, error) {
	objBytes, err := json.Marshal(obj)
	if err != nil {
		return nil, err
	}
	genericObj := map[string]interface{}{}
	if err := json.Unmarshal(objBytes, &genericObj); err != nil {
		return nil, err
	}
	for _, field := range fieldsToDrop {
		delete(genericObj, field)
	}
	return yaml.Marshal(genericObj) // or json.Marshal, depending on your needs.
}
```
Potentially this could be also done with some `reflect` usage to avoid the first `json.Marshal`, but I'm not sure which one would be more performant and how much we're concerned about the performance in this case.